Broken compile on non x86 machines

mkdir -p ".tmp" ".tmp/src"
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/attack.cpp -o .tmp/src/attack.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/bench.cpp -o .tmp/src/bench.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/cuckoo.cpp -o .tmp/src/cuckoo.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/eval.cpp -o .tmp/src/eval.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/history.cpp -o .tmp/src/history.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/init.cpp -o .tmp/src/init.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/io.cpp -o .tmp/src/io.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/magic.cpp -o .tmp/src/magic.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/main.cpp -o .tmp/src/main.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/makemove.cpp -o .tmp/src/makemove.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/misc.cpp -o .tmp/src/misc.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/move.cpp -o .tmp/src/move.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/movegen.cpp -o .tmp/src/movegen.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/movepicker.cpp -o .tmp/src/movepicker.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/nnue.cpp -o .tmp/src/nnue.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/perft.cpp -o .tmp/src/perft.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/piece_data.cpp -o .tmp/src/piece_data.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/position.cpp -o .tmp/src/position.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/random.cpp -o .tmp/src/random.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/search.cpp -o .tmp/src/search.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/threads.cpp -o .tmp/src/threads.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/time_manager.cpp -o .tmp/src/time_manager.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/ttable.cpp -o .tmp/src/ttable.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/tune.cpp -o .tmp/src/tune.o
g++ -funroll-loops -O3 -flto -fno-exceptions -std=gnu++2a -DNDEBUG -Wall -Wcast-qual -Wextra -Wshadow -Wdouble-promotion -Wformat=2 -Wnull-dereference -Wlogical-op -Wold-style-cast -Wundef -pedantic -DNETWORK_NAME=\"nn.net\" -DEVALFILE=\"nn.net\" -mcpu=apple-a14  -MMD -MP -c src/uci.cpp -o .tmp/src/uci.o
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
warning: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Wunknown-warning-option]
1 warning generated.
1 warning generated.
1 warning generated.
1 warning generated.
1 warning generated.
src/init.cpp:173:9: warning: variable 'count' set but not used [-Wunused-but-set-variable]
    int count = 0;
        ^
1 warning generated.
1 warning generated.
In file included from src/ttable.cpp:4:
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/xmmintrin.h:14:2: error: "This header is only meant to be used on x86 and x64 architecture"
#error "This header is only meant to be used on x86 and x64 architecture"
 ^
In file included from src/ttable.cpp:4:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/xmmintrin.h:17:
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:14:2: error: "This header is only meant to be used on x86 and x64 architecture"
#error "This header is only meant to be used on x86 and x64 architecture"
 ^
1 warning generated.
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:37:5: error: use of undeclared identifier '__builtin_ia32_emms'; did you mean '__builtin_isless'?
    __builtin_ia32_emms();
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/math.h:507:10: note: '__builtin_isless' declared here
  return __builtin_isless((type)__x, (type)__y);
         ^
In file included from src/ttable.cpp:4:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/xmmintrin.h:17:
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:37:25: error: too few arguments to function call, expected 2, have 0
    __builtin_ia32_emms();
    ~~~~~~~~~~~~~~~~~~~~^
1 warning generated.
1 warning generated.
1 warning generated.
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:54:19: error: use of undeclared identifier '__builtin_ia32_vec_init_v2si'
    return (__m64)__builtin_ia32_vec_init_v2si(__i, 0);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:71:12: error: use of undeclared identifier '__builtin_ia32_vec_ext_v2si'
    return __builtin_ia32_vec_ext_v2si((__v2si)__m, 0);
           ^
src/nnue.cpp:92:19: warning: unused variable 'REQUIRED_ITERS' [-Wunused-const-variable]
constexpr int32_t REQUIRED_ITERS = HIDDEN_SIZE / CHUNK_SIZE;
                  ^
1 warning generated.
src/position.cpp:509:16: warning: variable 'other' set but not used [-Wunused-but-set-variable]
    ZobristKey other = (originalKey ^ OldKey(1));
               ^
1 warning generated.
2 warnings generated.
1 warning generated.
2 warnings generated.
1 warning generated.
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:133:19: error: use of undeclared identifier '__builtin_ia32_packsswb'
    return (__m64)__builtin_ia32_packsswb((__v4hi)__m1, (__v4hi)__m2);
                  ^
1 warning generated.
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:163:19: error: use of undeclared identifier '__builtin_ia32_packssdw'
    return (__m64)__builtin_ia32_packssdw((__v2si)__m1, (__v2si)__m2);
                  ^
1 warning generated.
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:193:19: error: use of undeclared identifier '__builtin_ia32_packuswb'
    return (__m64)__builtin_ia32_packuswb((__v4hi)__m1, (__v4hi)__m2);
                  ^
1 warning generated.
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:220:19: error: use of undeclared identifier '__builtin_ia32_punpckhbw'
    return (__m64)__builtin_ia32_punpckhbw((__v8qi)__m1, (__v8qi)__m2);
                  ^
1 warning generated.
1 warning generated.
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:243:19: error: use of undeclared identifier '__builtin_ia32_punpckhwd'
    return (__m64)__builtin_ia32_punpckhwd((__v4hi)__m1, (__v4hi)__m2);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:264:19: error: use of undeclared identifier '__builtin_ia32_punpckhdq'
    return (__m64)__builtin_ia32_punpckhdq((__v2si)__m1, (__v2si)__m2);
                  ^
1 warning generated.
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:291:19: error: use of undeclared identifier '__builtin_ia32_punpcklbw'
    return (__m64)__builtin_ia32_punpcklbw((__v8qi)__m1, (__v8qi)__m2);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:314:19: error: use of undeclared identifier '__builtin_ia32_punpcklwd'
    return (__m64)__builtin_ia32_punpcklwd((__v4hi)__m1, (__v4hi)__m2);
                  ^
2 warnings generated.
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:335:19: error: use of undeclared identifier '__builtin_ia32_punpckldq'
    return (__m64)__builtin_ia32_punpckldq((__v2si)__m1, (__v2si)__m2);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:356:19: error: use of undeclared identifier '__builtin_ia32_paddb'; did you mean '__builtin_arm_addg'?
    return (__m64)__builtin_ia32_paddb((__v8qi)__m1, (__v8qi)__m2);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:37:5: note: '__builtin_arm_addg' declared here
    __builtin_ia32_emms();
    ^
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:356:19: error: first argument of MTE builtin function must be a pointer ('__v8qi' (vector of 8 'char' values) invalid)
    return (__m64)__builtin_ia32_paddb((__v8qi)__m1, (__v8qi)__m2);
                  ^                    ~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:377:19: error: use of undeclared identifier '__builtin_ia32_paddw'; did you mean '__builtin_arm_addg'?
    return (__m64)__builtin_ia32_paddw((__v4hi)__m1, (__v4hi)__m2);
                  ^
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:37:5: note: '__builtin_arm_addg' declared here
    __builtin_ia32_emms();
    ^
/Library/Developer/CommandLineTools/usr/lib/clang/15.0.0/include/mmintrin.h:377:19: error: first argument of MTE builtin function must be a pointer ('__v4hi' (vector of 4 'short' values) invalid)
    return (__m64)__builtin_ia32_paddw((__v4hi)__m1, (__v4hi)__m2);
                  ^                    ~~~~~~~~~~~~
fatal error: too many errors emitted, stopping now [-ferror-limit=]
1 warning and 20 errors generated.
make: *** [.tmp/src/ttable.o] Error 1

--- a/src/ttable.cpp
+++ b/src/ttable.cpp
@@ -1,8 +1,10 @@
 #include "ttable.h"
-#include <cassert>
 #include "io.h"
-#include "xmmintrin.h"
 #include <iostream>
+// This include breaks on non x86 target platforms
+#if defined(__INTEL_COMPILER) || defined(_MSC_VER)
+#include "xmmintrin.h"
+#endif
 
 TTable TT;
 

